src/TorchSharp/TorchVision/Image.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+
+using static TorchSharp.torch;
+
+namespace TorchSharp.torchvision
+{
+    public static partial class io
+    {
+
+        public static IImager DefaultImager { get; set; } = new ImagerSharp();
+
+        public enum ImageFormat
+        {
+            PNG,
+            JPEG,
+            UNKNOWN
+        }
+        public interface IImager
+        {
+            ImageFormat DetectFormat(byte[] image);
+            Tensor DecodeImage(byte[] image, ImageReadMode mode = ImageReadMode.UNCHANGED);
+            Tensor DecodeImage(byte[] image, ImageFormat format, ImageReadMode mode = ImageReadMode.UNCHANGED);
+            byte[] EncodeImage(Tensor image, ImageFormat format);
+        }
+
+        public enum ImageReadMode
+        {
+            UNCHANGED,
```

Comment:
>UNCHANGED [](http://example.com/codeflow?start=12&length=9)

Maybe using C# style casing will be better here? i.e. Unchanged, Gray, GrayAlpha, Rgb...etc.

Reply:
`ImageReadMode` and its options were written to resemble their [pytorch version](https://pytorch.org/vision/stable/generated/torchvision.io.ImageReadMode.html#torchvision.io.ImageReadMode) but if it is decided that we should change those then we should probably also change the function names as well.

Reply:
Generally, I'm in favor of keeping the Pytorch casing wherever it enables 'copy-and-paste' from Python. If we're replacing a string with an enum, it's not as clear.

Reply:
`ImageReadMode` is an enum in pytorch.

Reply:
Ok then to keep the same casing as pytorch then.